File: 3. Python/2. Silver/4673.py
def d(n):
    sum = 0
    sum += n
    temp = n
    while temp > 0:
        sum += temp % 10
        temp = temp // 10
    return sum

num = [i for i in range(1, 10001)]

for i in range(1, 10001):
    if d(i) in num:
        num.remove(d(i))
        
for i in num:
    print(i)

File: 1. C, C++/2. Silver/4673.c
#include <stdio.h>
int d(int n) {
	int sum = n;
	int temp = n;

	while (temp > 0) {
		sum += temp % 10;
		temp = temp / 10;
	}
	return sum;
}

int main(void) {
	int num[10001];
	for (int i = 0; i < 10001; i++)
		num[i] = i;

	for (int i = 1; i <= 10000; i++) {
		if (d(i) <= 10000)
			num[d(i)] = 0;
	}

	for (int i = 0; i < 10001; i++) {
		if (num[i] != 0)
			printf("%d\n", num[i]);
	}
}